src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy

```diff
@@ -3,9 +3,36 @@ package com.avioconsulting.mule.linter.model
 class Application {
 
     File applicationPath
+    Map<String,ProjectFile> files = new HashMap<String,ProjectFile>()
 
     Application(File applicationPath) {
         this.applicationPath = applicationPath
+        if( !this.applicationPath.exists()) {
+            throw new FileNotFoundException(applicationPath.absolutePath, "Application directory does not exists.")
+        }
+
+        //Parse out the files necessary for an application.
+        files.put("POM", new File(applicationPath, "pom.xml"))
```

Comment:
if files is a Map<String,ProjectFile> why are you putting a File directly into it and not a ProjectFile?